pkg/components/util/helm.go

```diff
@@ -34,23 +35,16 @@ import (
 	"github.com/kinvolk/lokomotive/pkg/k8sutil"
 )
 
-// LoadChartFromAssets takes in an asset location and returns a Helm
-// Chart object or an error.
-func LoadChartFromAssets(location string) (*chart.Chart, error) {
-	tmpDir, err := ioutil.TempDir("", "lokoctl-chart-")
-	if err != nil {
-		return nil, errors.Wrap(err, "creating temporary dir")
-	}
-	defer os.RemoveAll(tmpDir)
+// Chart finds a chart named name in the assets and returns a pointer to a chart.Chart struct.
+func Chart(name string) (*chart.Chart, error) {
```

Comment:
If we touch that, maybe we can introduce it in `pkg/components` package?

Also, I find a simple `Chart` name a bit ambiguous. Does it imply `GetChart` ? And why do we remove `FromAssets` from the name? Do we plan to load charts from some other place in the future (e.g. Helm repository)?

I think having more fine-grained commits would allow including exactly this kind of information with reasoning behind made decisions.

Reply:
>If we touch that, maybe we can introduce it in pkg/components package?

If we move this around I'd rather make this a generic function which can be used to extract *any* chart, never mind if it's a component or a control plane chart. We can remove `assets.ComponentsSource` from the function's implementation and instead construct the source path outside before calling `Chart()`.

>Also, I find a simple Chart name a bit ambiguous. Does it imply GetChart ?

It does imply "get chart". What is your other interpretation?

Go getters usually don't have "Get" in their name: https://golang.org/doc/effective_go.html#Getters

>If you have a field called `owner` (lower case, unexported), the getter method should be called `Owner` (upper case, exported), not `GetOwner`.

>And why do we remove FromAssets from the name? Do we plan to load charts from some other place in the future (e.g. Helm repository)?

How about we move `Chart()` into `assets`? You would then have:

```go
// Get a control plane chart.
p := filepath.Join(assets.ControlPlaneSource, "kubelet")
c, err := assets.Chart(p)
if err != nil {...}

// Get a component chart.
p := filepath.Join(assets.ComponentsSource, "contour")
c, err := assets.Chart(p)
if err != nil {...}
```

Reply:
Actually it might be nicer to have two wrapper functions which use the same unexported chart extraction function: `ComponentChart()` and `ControlPlaneChart()`.

Reply:
>It does imply "get chart". What is your other interpretation?

>Go getters usually don't have "Get" in their name: https://golang.org/doc/effective_go.html#Getters

I didn't know that, thanks! It totally make sense :) 

>If we move this around I'd rather make this a generic function which can be used to extract any chart, never mind if it's a component or a control plane chart. We can remove assets.ComponentsSource from the function's implementation and instead construct the source path outside before calling Chart().

>How about we move Chart() into assets? You would then have:

Hmm, I'd rather keep `assets` package minimal. Perhaps it would make sense to have `pkg/helm` package then (or `internal/helm` :wink:) to keep there all Helm-related functionality, including extraction of charts from assets.

This package could also handle chart installation, so we don't keep this in Components package. Components package would only consume some kind of installer/manager interface (which helm package would implement). This way, the composition of those concepts would be happening on cluster management level (so helm with components and system charts and assets etc.). Hopefully it's clear what I mean. What do you think?

>Actually it might be nicer to have two wrapper functions which use the same unexported chart extraction function: ComponentChart() and ControlPlaneChart().

:+1: for helpers.

Reply:
Yes, I thought about creating a new Helm package but didn't want to digress too much from the scope of this PR. I can take a quick look at it to see if it's trivial. If not, I'd be fine with letting the two helper functions live in `assets` for now and we can move them to `helm` later on.

Reply:
I've created `pkg/helm` and moved the logic discussed above there. I did **not** move all Helm-related logic to this package - let's do this gradually in subsequent PRs.

Reply:
Opened https://github.com/kinvolk/lokomotive/issues/819 for follow up.